src/models/event-timeline-set.ts

```diff
@@ -119,13 +120,15 @@ export class EventTimelineSet extends TypedEventEmitter<EmittedEvents, EventTime
      * Set to true to enable improved timeline support.
      * @param {Object} [opts.filter = null]
      * The filter object, if any, for this timelineSet.
-     * @param {MatrixClient} client the Matrix client which owns this EventTimelineSet,
+     * @param {MatrixClient=} client the Matrix client which owns this EventTimelineSet,
      * can be omitted if room is specified.
+     * @param {Thread=} thread the thread to which this timeline set relates.
      */
     constructor(
         public readonly room: Room | undefined,
         opts: IOpts = {},
         client?: MatrixClient,
+        public readonly thread?: Thread,
```

Comment:
Seems like we should put this stuff in `opts`

Reply:
> Seems like we should put this stuff in opts

Why? https://github.com/vector-im/element-meta/pull/216/files doesn't favour it in any way shape or form

Reply:
Feel free to defer in favor of the style-guide.

The function signature here is getting pretty complex and we already have `opts` where this seems like it would fit.